res/ai/CommonAI/AIEngine.lua

```diff
@@ -349,6 +349,7 @@ _G["AITask"] = class(KIDataObjekt, function(c)
 	c.MaxTicks = 30 --Wie viele Ticks darf der Task maximal laufen?
 	c.IdleTicks = 10 --Wie viele Ticks soll nichts gemacht werden?
 	c.TargetRoom = -1 -- Wie lautet die ID des Standard-Zielraumes? !!! Muss überschrieben werden !!!
+	c.TargetID = -1 -- Wie lautet die ID des "Ziels" (Tuer, Hotspot,...) zum Zielraum? !!! Muss überschrieben werden !!!
```

Comment:
Genau so eine Dopplung hätte ich eigentlich gern vermieden. Eine Option wäre, dass nur der GoToRoom-Job selbst prüft, ob die TargetRoom-ID als Ergebnis NotFound liefert und dann statt GoToRoom GoToTarget ausführt.

Könntest Du Dir die Änderungen in AIEngine.lua lokal vorhalten und für den PR rückgängig machen? Ich würde mich dann nochmal an einer alternativen, "transparenten" Engineanpassung versuchen.

Reply:
Das ist hier nur ein Zwischenschritt - normalerweise wuerde ich "TargetRoom" umbenennen in "TargetRoomID", das kann man aber ja nicht so ohne "Zwischenschritt". Gleichzeitig ist "TargetRoom(ID)" notwendig, wenn man wissen will, ob ein Raum gesperrt ist - aber bspweise die Tuer 2 von 2 erreichbar (TargetID statt RoomID).

Reply:
> Dopplung

Es ist eben keine Dopplung. TargetRoom beschreibt den RAUM. TargetID den "Zugang" zum Raum (Tuer, Hotspot ...).

Reply:
OK ungünstig formuliert. Ich hätte am Ende gern nur eine Variable, da es sonst Inkonsistenzen geben kann. (TargetRaum im Task mit Office belegt, TargetId der Tür zu Studio von anderem Spieler; welche ID soll die Engine nun für die Navigation nutzen).

Ich denke auch nicht dass es zukünftig Wurmlöcher geben soll (zwei Türen für ein Studio in unterschiedlichen Etagen). Typischerweise wird man in einen Raum wollen, in Spezialfällen zu einem anderen Target.